src/Migrate.php

```diff
@@ -0,0 +1,795 @@
+    {
+        $user = $mainRole = $mainRoleWithGrants['name'];
+
+        $this->destinationConnection->createRole(['name' => $mainRole, 'privilege' => 'OWNERSHIP']);
+
+        $mainRoleGrants = [
+            'GRANT CREATE DATABASE ON ACCOUNT TO ROLE %s;',
+            'GRANT CREATE ROLE ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+            'GRANT CREATE USER ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+        ];
+
+        foreach ($mainRoleGrants as $mainRoleGrant) {
+            $this->destinationConnection->query(sprintf($mainRoleGrant, $mainRole));
+        }
+
+        $warehouses = array_filter($mainRoleWithGrants['assignedGrants'], fn($v) => $v['granted_on'] === 'WAREHOUSE');
+        $useWarehouse = false;
+        foreach ($warehouses as $warehouse) {
+            $warehouseSize = self::createWarehouse($warehouse);
+            $this->destinationConnection->assignGrantToRole($warehouse);
+
+            if ($useWarehouse === false || $warehouseSize === 'X-Small') {
+                $useWarehouse = $warehouse;
+            }
+        }
+
+        $this->destinationConnection->query(sprintf(
+            'USE WAREHOUSE %s',
+            QueryBuilder::quoteIdentifier($useWarehouse['name'])
+        ));
+
+        $this->destinationConnection->query(sprintf(
+            'CREATE USER %s PASSWORD=%s DEFAULT_ROLE=%s',
+            $user,
+            QueryBuilder::quote($users[$user]),
+            $mainRole
+        ));
+
+        $this->destinationConnection->query(sprintf(
+            'GRANT ROLE %s TO USER %s;',
+            $mainRole,
+            $user
+        ));
+
+        $projectUsers = array_filter($mainRoleWithGrants['assignedGrants'], function ($v) use ($databases) {
+            if ($v['privilege'] !== 'OWNERSHIP') {
+                return false;
+            }
+            if ($v['granted_on'] !== 'USER') {
+                return false;
+            }
+            return in_array($v['name'], $databases);
+        });
+
+        foreach ($projectUsers as $projectUser) {
+            self::createUser($projectUser, $users);
+            $this->destinationConnection->assignGrantToRole($projectUser);
+        }
+
+        $this->destinationConnection->useRole('ACCOUNTADMIN');
+
+        $this->destinationConnection->query(sprintf(
+            'GRANT ROLE %s TO ROLE SYSADMIN;',
+            $mainRole
+        ));
+    }
+
+    public function cleanupProject(): void
+    {
+        $this->destinationConnection->useRole('ACCOUNTADMIN');
+
+        $databases = [
+            'SAPI_9472_OLD',
+            'SAPI_9473_OLD',
+        ];
+        foreach ($databases as $database) {
+            $this->destinationConnection->query(sprintf(
+                'DROP DATABASE IF EXISTS %s;',
+                QueryBuilder::quoteIdentifier($database)
+            ));
+        }
+
+        $warehouses = [
+            'MIGRATE',
+            'MIGRATE_SMALL',
+            'MIGRATE_MEDIUM',
+            'MIGRATE_LARGE',
+        ];
```

Comment:
Tohle jsou nějaké speciální warehousy co si uděláš? Když je dropneš tady per-projekt, tak ztratíš všechny granty na ně i v jiných projektech. To je feature?

Reply:
sakra, sry to je metoda jen pro moje testování (napíšu to do komentáře) - celou jí smažu až to bude hotové.... na promazání dat je co sem linkoval na slacku
https://github.com/keboola/project-migration-tool/blob/ondra-init/src/Migrate.php#L597
warehouse by se určitě mazat neměly, stejně jako ta main role/user (KEBOOLA_STORAGE)